### ISSUE TITLE
Response emitter not setting Content-Length

### ISSUE THREAD

**Issue by mnapoli** [zendframework/zend-diactoros#84]
This is more of a question about the response emitters than a bug report.

Slim does this processing before emitting the response: https://github.com/slimphp/Slim/blob/3.x/Slim/App.php#L472-L484

Should this be added to Diactoros, or is it expected to be done by the consuming project (i.e. me)?

**Comment by weierophinney** [comment]
That's a great catch; I think it *should* be the job of the emitter, to be honest. Considering the information is in the response stream already (via `getSize()`), it could be something like this:

```php
if (! $response->hasHeader('Content-Length')) {
    $response = $response->withHeader('Content-Length', $response->getBody()->getSize());
}
```

That could even be provided as a trait method (e.g., `injectContentLength(ResponseInterface $response): ResponseInterface`), which emitters can then compose in order to provide this functionality.

Want to give it a whirl, @mnapoli ?

**Comment by mnapoli** [comment]
Great, that will also simplify my code. It's on my todo list then!

**Comment by mnapoli** [comment]
This was fixed by #88.

**Comment by franzliedke** [comment]
A little late to the game, but why exactly was this added? AFAIK, `Content-Length` is only *needed* for downloads.

I am asking because we [have some issue reports about truncated JSON responses for Flarum](https://github.com/flarum/core/issues/825) which seem to have to do with this header not matching the true content length.
